ipatests/test_xmlrpc/test_host_plugin.py

```diff
@@ -337,6 +341,87 @@ def test_add_host_with_null_password(self, host):
         host.track_create()
         host.check_create(result)
 
+    @staticmethod
+    def modify_config_maxhostname(host_tracker, value):
+        command = host_tracker.make_command(
+            'config_mod',
+            **dict(
+                setattr=u'ipamaxhostnamelength={}'.format(value)))
+        command()
+
+    @staticmethod
+    def generate_hostname(total_length):
+        if total_length < 9:
+            raise ArithmeticError("Total DNS length in theses tests"
+                                  "must be at least 9")
+        return "aaa.bbb.{}".format((total_length - 8) * "c")
+
+    def test_config_maxhostname_invalid(self):
+        """Change config maxhostname to an invalid value
+        (lower than MAXHOSTNAMELEN). Should fail"""
+        testhost = HostTracker(name=u'testshost')
+        with raises_exact(errors.ValidationError(
+                name='ipamaxhostnamelength',
+                error='must be at least {}'.format(MAXHOSTNAMELEN))):
+            self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN // 2)
+
+    def test_raise_hostname_limit_above_maxhostnamelimit(self):
+        """Raise hostnamelimit above default. Should pass"""
+        testhost = HostTracker(name=u'testshost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN * 2)
+
+    def test_try_hostname_length_above_maxhostnamelimit(self):
+        """Try to create host with hostname length above
+        hostnamelength limit. Should fail"""
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN)
+        host_create_command = testhost.make_create_command()
+        with raises_exact(errors.ValidationError(
+                name=u'ipamaxhostnamelength',
+                error=u'can be at most {} characters'.format(
+                    MAXHOSTNAMELEN - 1))):
+            host_create_command()
+
+    def test_try_hostname_length_bellow_maximum(self):
+        """Try to create host with valid hostname. Should pass"""
+        valid_length = MAXHOSTNAMELEN // 2
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(valid_length)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN)
+        host_create_command = testhost.make_create_command()
+        host_create_command()
+        assert host_create_command.returncode == 0
```

Comment:
Make sure to remove the host entry at the end of the test, you can use test_host.ensure_missing() IIRC